pkg/process/extract.go

```diff
@@ -114,25 +117,57 @@ func (t trace) Name() string {
 
 // ErrorPrimitiveReached occurs when walkJSON reaches the end of nested dicts without finding a valid Kubernetes manifest
 type ErrorPrimitiveReached struct {
-	path, key string
-	primitive interface{}
+	path, key        string
+	primitive        interface{}
+	containingObj    objx.Map
+	containingObjErr error
+}
+
+func (e ErrorPrimitiveReached) WithContainingObj(obj objx.Map, err error) ErrorPrimitiveReached {
+	if e.containingObj == nil {
+		e.containingObj = obj
+		e.containingObjErr = err
+	}
+	return e
 }
 
 func (e ErrorPrimitiveReached) Error() string {
-	return fmt.Sprintf("recursion did not resolve in a valid Kubernetes object. "+
-		" In path `%s` found key `%s` of type `%T` instead.",
-		e.path, e.key, e.primitive)
+	container, _ := yaml.Marshal(e.containingObj)
+
+	return fmt.Sprintf(`recursion ended on key %q of type %T which does not belong to a valid Kubernetes object
+instead, it an attribute of the following object (at path %s):
```

Comment:
I find this wording a little bit unclear. "recursion ended" doesn't mean all that much to me. Would it be correct to say that `e.key` was the last _okay_ thing in the object somehow? Like "the last object successfully validated was %q of type %T, the error is probably just after that"?

And maybe instead of "instead, it an attribute of the following object" it could say something like "this is the invalid object (at path %s): <...>"?

Reply:
The error is found when recursion ends up on a field that is a primitive type and not part of a Kubernetes object. So if it said:

```
found field not belonging to a Kubernetes object
field "%q" of type "%T" is part of this object (at path %s):
...

this object is not a valid Kubernetes object because: ...
```

Would that be better? Not sure how to phrase this, it's not correct to say that "the last object successfully validated was %q of type %T"

Reply:
what _is_ `e.key` and `e.primitive` at this point? I'm trying to understand what it's helpful to say about it to a user seeing this error. It seems a bit random to me if it's not the last thing we managed to validate - maybe leave it out if there's no really decent way to explain how to make use of that info?

anyway it's not a huge deal, your new proposal looks better - feel free to go with it!

Reply:
What about:

```found invalid Kubernetes object (at <path>): <reason>

<yaml>
```

Nothing about the primitive, which is useless indeed. Thinking about users who have no knowledge of the validation algorithm

Reply:
exactly, that's me right now. sounds great :+1: 

you could include it in debug mode if it's helpful for developers (but probably there would be other/better places to output the same info)